homeassistant/components/abode/alarm_control_panel.py

```diff
@@ -9,22 +9,24 @@
     STATE_ALARM_DISARMED,
 )
 
-from . import ATTRIBUTION, DOMAIN as ABODE_DOMAIN, AbodeDevice
+from . import AbodeDevice
+from .const import DOMAIN, ATTRIBUTION
 
 _LOGGER = logging.getLogger(__name__)
 
 ICON = "mdi:security"
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up an alarm control panel for an Abode device."""
-    data = hass.data[ABODE_DOMAIN]
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
 
-    alarm_devices = [AbodeAlarm(data, data.abode.get_alarm(), data.name)]
 
-    data.devices.extend(alarm_devices)
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up an alarm control panel for an Abode device."""
+    data = hass.data[DOMAIN]
 
-    add_entities(alarm_devices)
+    async_add_devices([AbodeAlarm(data, data.abode.get_alarm(), data.name)], True)
```

Comment:
Does `data.abode.get_alarm()` do I/O?

Reply:
It appears to do a `get` request. From abodepy:

`response = self.send_request("get", CONST.DEVICES_URL)`

Reply:
Then we need to schedule it on the executor thread pool. Use `await hass.async_add_executor_job(...)`.